webui/src/js/viewModels/model-design-view.js

```diff
@@ -3,81 +3,241 @@
  * Copyright (c) 2021, 2022, Oracle and/or its affiliates.
  * Licensed under The Universal Permissive License (UPL), Version 1.0 as shown at https://oss.oracle.com/licenses/upl/
  */
-define(['accUtils', 'utils/i18n', 'knockout', 'models/wkt-project', 'utils/url-catalog', 'utils/wkt-logger',
-  'ojs/ojinputtext', 'ojs/ojlabel', 'ojs/ojbutton', 'ojs/ojformlayout' ],
-function(accUtils, i18n, ko, project, urlCatalog) {
-  function ModelDesignViewModel() {
 
-    this.connected = () => {
-      accUtils.announce('Model design view loaded.', 'assertive');
-      // Implement further logic if needed
-    };
+define(['accUtils', 'utils/i18n', 'knockout', 'models/wkt-project', 'utils/url-catalog', 'utils/wkt-logger', 'wrc-frontend/core/parsers/yaml', 'wrc-frontend/integration/viewModels/utils',
+        'wdt-model-designer/loader', 'ojs/ojinputtext', 'ojs/ojlabel', 'ojs/ojbutton', 'ojs/ojformlayout'],
+  function(accUtils, i18n, ko, project, urlCatalog, wktLogger, YamlParser, ViewModelUtils) {
+    function ModelDesignViewModel() {
```

Comment:
hmm...  We already are pulling in `js-yaml` and using it everywhere, is there something fundamentally different about `wrc-frontend/core/parsers/yaml`?  I am concerned because using 2 different parsers may result in subtle bugs caused by parser differences.  Seems like we should use the same parser...

Reply:
We used the yaml parser in wrc-jet-pack, because we didn't know which WKT-UI module you guys use to parse yaml. Switching to use whatever the WKT-UI is using should the relatively easy to do.